strawberry/schema/schema_converter.py

```diff
@@ -362,9 +362,16 @@ def from_input_field(
             ),
         )
         default_value: object
-
+        model_dump = getattr(field.default_value, "model_dump", None)
+        is_not_type = not isinstance(field.default_value, type)
         if field.default_value is UNSET or field.default_value is dataclasses.MISSING:
             default_value = Undefined
+        elif dataclasses.is_dataclass(field.default_value) and is_not_type:
+            dataclass_instance = cast(Any, field.default_value)
```

Comment:
I was fighting with mypy here.  The dataclasses.asdict method only takes dataclass instances.  Mypy thought we still had a dataclass instance or a dataclass type, however, I'm specifically checking to see if it's a type or not with `is_not_type`.

Reply:
I'm guessing I will have to pull this out anyway.  I was hoping to solve this for dataclasses and pydantic models, but it looks like newer versions of python are more strict on what values can be set as the default value for a field on a dataclass.  The new tests I added are failing in python 3.11+.